task_18/toma.lobanova/src/main/java/com/hillel/linkedlist/CustomLinkedList.java

```diff
@@ -0,0 +1,172 @@
+package com.hillel.linkedlist;
+
+import java.util.EmptyStackException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.StringJoiner;
+
+public class CustomLinkedList implements Iterable {
+  Node first;
+  int count;
+  Node last;
+
+  public void add(int data) {
+    Node node = new Node(data);
+    if (last == null) {
+      first = node;
+      last = node;
+    } else {
+      last.next = node;
+      last = node;
+    }
+    count++;
+  }
+
+  public int get(int position) {
+    int currentIndex = 1;
+    Node current = first;
+    if (current == null) {
+      throw new EmptyStackException();
+    }
+    if (position < 0 || position >= count) {
+      throw new IllegalArgumentException();
+    }
+    while (currentIndex <= position) {
+      currentIndex++;
+      current = current.next;
+    }
+
+    return current.data;
+  }
+
+  public void remove() {
+    if (first != null) {
+      Node temp = first.next;
+      first = temp;
+      count--;
+    } else {
+      throw new EmptyStackException();
+    }
+    if (count == 0) {
+      last = null;
+    }
+  }
+
+  public void introduceCycle(int index) {
+    Node current = first;
+    for (int i = 1; i <= index; i++) {
+      current = current.next;
+    }
+    last.next = current;
+  }
+
+  public boolean hasCycle() {
+    Node current = first;
+    for (int i = 1; i < count; i++) {
```

Comment:
Forgot to mention that for this function we assume that the number of elements is unknown. That would be to easy ;)